lib/workers/repository/update/branch/index.ts

```diff
@@ -114,19 +126,37 @@ export async function processBranch(
   const artifactErrorTopic = emojify(':warning: Artifact update problem');
   try {
     // Check if branch already existed
-    const existingPr = branchPr ? undefined : await prAlreadyExisted(config);
+    const existingPr = await prAlreadyExisted(config);
     if (existingPr && !dependencyDashboardCheck) {
-      logger.debug(
-        { prTitle: config.prTitle },
-        'Closed PR already exists. Skipping branch.'
-      );
-      await handlepr(config, existingPr);
-      return {
-        branchExists: false,
-        prNo: existingPr.number,
-        result: BranchResult.AlreadyExisted,
-      };
+      // Recreates pr if merged pr already exists
+      if (existingPr.state === 'merged') {
+        if (config.automerge) {
+          logger.debug(
+            'Disable automerge to prevent re-created PRs from being merged automatically.'
+          );
+        }
+
+        config = { ...config, ...recreateMergedPrConfig(branchPr) };
+        if (config.recreateMergedPr) {
```

Comment:
```suggestion
        if (!branchPr) {
```

I think this is easier logic to follow

Reply:
having the recreateMergedPr = true seems more informative and we can't delete it totally so good to use it 🧒